quarkus-picocli/src/test/java/PicocliDevIT.java

```diff
@@ -0,0 +1,102 @@
+import static java.util.concurrent.CompletableFuture.runAsync;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.quarkus.ts.qe.command.AgeCommand;
+import io.quarkus.ts.qe.command.EntryCommand;
+import io.quarkus.ts.qe.command.HelloCommand;
+import io.quarkus.ts.qe.configuration.Config;
+import io.quarkus.ts.qe.services.AgeService;
+import io.quarkus.ts.qe.services.HelloService;
+
+@QuarkusScenario
+public class PicocliDevIT {
+
+    @QuarkusApplication(classes = { EntryCommand.class, Config.class, HelloCommand.class, AgeCommand.class, AgeService.class,
+            HelloService.class }, properties = "dev.properties")
+    static final RestService greetingApp = new RestService()
+            .withProperty("quarkus.args", "greeting --name QE")
+            .setAutoStart(false);
+
+    @QuarkusApplication(classes = { EntryCommand.class, Config.class, HelloCommand.class, AgeCommand.class, AgeService.class,
+            HelloService.class }, properties = "dev.properties")
+    static final RestService ageApp = new RestService()
+            .withProperty("quarkus.args", "age --age 30")
+            .setAutoStart(false);
+
+    @QuarkusApplication(classes = { EntryCommand.class, Config.class, HelloCommand.class, AgeCommand.class, AgeService.class,
+            HelloService.class }, properties = "dev.properties")
+    static final RestService greetingBlankArgumentApp = new RestService()
+            .withProperty("quarkus.args", " --name QE")
+            .setAutoStart(false);
+
+    @QuarkusApplication(classes = { EntryCommand.class, Config.class, HelloCommand.class, AgeCommand.class, AgeService.class,
+            HelloService.class }, properties = "dev.properties")
+    static final RestService greetingInvalidArgumentApp = new RestService()
+            .withProperty("quarkus.args", "greeting -x QE")
+            .setAutoStart(false);
+
+    @QuarkusApplication(classes = { EntryCommand.class, Config.class, HelloCommand.class, AgeCommand.class, AgeService.class,
+            HelloService.class }, properties = "dev.properties")
+    static final RestService bothTopCommandApp = new RestService()
+            .setAutoStart(false);
+
+    @Test
+    public void verifyErrorForApplicationScopedBeanInPicocliCommand() {
+        try {
+            runAsync(ageApp::start);
```

Comment:
Test is called `verify error`, and test plans say, that there should be an error for application scoped commands, yet there is no error.

Reply:
good catch, I've just fixed.